scripts/dragen_reprocessing/get_dragen_metrics.py

```diff
@@ -0,0 +1,149 @@
+            'metric_type': 'MAPPING/ALIGNING SUMMARY'
+        },
+        'Not properly paired reads (discordant)': {
+            'tdr_name': CHIMERA_RATE,
+            'metric_type': 'MAPPING/ALIGNING SUMMARY'
+        },
+        'Mapped reads': {
+            'tdr_name': MAPPED_READS,
+            'metric_type': 'MAPPING/ALIGNING SUMMARY'
+        }
+    }
+}
+VC_METRICS = {
+    'file_name': '{sample}.vc_metrics.csv',
+    'metrics': {
+        'Percent Callability': {
+            'tdr_name': PERCENT_CALLABILITY,
+            'metric_type': 'VARIANT CALLER POSTFILTER'
+        }
+    }
+}
+
+
+def get_args() -> Namespace:
+    # Set up argument parser
+    parser = ArgumentParser(description="Get dragen metrics from output metric files")
+    parser.add_argument('-o', '--output_path', required=True, help="Path to dragen output files")
+    parser.add_argument("-s", "--sample_name", required=True, help="Sample name")
+    return parser.parse_args()
+
+
+class GetMetricsFilesContents:
+    def __init__(self, output_path: str, sample_name: str):
+        self.output_path = output_path
+        self.sample_name = sample_name
+        self.gcs_client = storage.Client()
+
+    def read_file(self, metrics_file: str) -> list[str]:
+        full_gcp_path = os.path.join(
+            self.output_path, metrics_file.format(sample=self.sample_name)
+        )
+        bucket_name, blob_name = full_gcp_path.split('/')[2], '/'.join(full_gcp_path.split('/')[3:])
+        bucket = self.gcs_client.get_bucket(bucket_name)
+        blob = bucket.blob(blob_name)
+        logging.info(f"Reading {full_gcp_path}")
+        return blob.download_as_string().decode().split('\n')
+
+    def get_metrics_files(self) -> tuple[list[str], list[str], list[str], list[str]]:
+        mapping_metrics_contents = self.read_file(MAPPING_METRICS['file_name'])
+        q1_coverage_reports_contents = self.read_file(Q1_COVERAGE['file_name'])
+        q3_coverage_reports_contents = self.read_file(Q3_COVERAGE['file_name'])
+        vc_metrics_contents = self.read_file(VC_METRICS['file_name'])
+        return mapping_metrics_contents, q1_coverage_reports_contents, q3_coverage_reports_contents, vc_metrics_contents
+
+
+class GetMetrics:
+    def __init__(self, metrics_contents: list[str], metrics_dict: dict):
+        self.metrics_contents = metrics_contents
+        self.metrics_dict = metrics_dict
+
+    def get_metrics(self) -> dict:
+        metrics_dict = {}
+        for line in self.metrics_contents:
+            if line:
+                split_line = line.split(',')
+                metric_type = split_line[0]
+                metric_key = split_line[2]
+                value = split_line[3]
+                expected_metric_dict = self.metrics_dict.get(metric_key)
+                if expected_metric_dict:
+                    if expected_metric_dict['metric_type'] == metric_type:
+                        metrics_dict[expected_metric_dict['tdr_name']] = value
```

Comment:
I see the error is thrown later but should we log an error or warning in case the expected value isn't present in the file before we make the final metrics file? or you want it to continue anyways?